src/Agent.Worker/ExecutionContext.cs

```diff
@@ -578,6 +580,24 @@ private void JobServerQueueThrottling_EventReceived(object sender, ThrottlingEve
             if (!_throttlingReported)
             {
                 this.Warning(StringUtil.Loc("ServerTarpit"));
+
+                if (this.Variables.System_TFCollectionUrl != null)
+                {
+                    // Construct a URL to the resource utilization page, to aid the user debug throttling issues
+                    UriBuilder uriBuilder = new UriBuilder(Variables.System_TFCollectionUrl);
+                    NameValueCollection query = HttpUtility.ParseQueryString(uriBuilder.Query);
+                    DateTime endTime = DateTime.UtcNow;
+                    string queryDate = endTime.AddHours(-1).ToString("s") + "," + endTime.ToString("s");
```

Comment:
i am not sure endtime - 1 hour is accurate, you can take a timestamp during ExecutionContext.InitailJob

Reply:
I was told to keep the query time range scoped to 1 hour. Otherwise, the report process can be expensive and take a longer time to load the page.

It can be safely assumed that the load pattern that caused the server to throttle occurred within that past hour. The RU service queues a job to evaluate thresholds once every minute, and the windows to calculate the threshold violations are 5 minutes and 1 hour, respectively.